File: day17/characterCount.cpp
#include <iostream>
#include <string>

int main() {
    std::string input;
    char target;

    // Get non-empty sentence from user
    while (true) {
        std::cout << "Please enter a sentence:\n";
        std::getline(std::cin, input);

        if (!input.empty()) break;
        std::cout << "\nPlease enter at least one character.\n\n";
    }

    // Get character to search for
    std::cout << "\nPlease enter a character to count:\n";
    std::cin >> target;

    // Count occurrences
    int count = 0;
    for (char c : input) {
        if (c == target) count++;
    }

    std::cout << "\nThe character '" << target << "' appears " << count << " time"
              << (count == 1 ? "." : "s.") << std::endl;

    return 0;
}

File: day17/character_count.rs
use std::io::{self, Write};

fn main() {
    let sentence = loop {
        print!("Please enter a sentence:\n");
        io::stdout().flush().unwrap();

        let mut input = String::new();
        io::stdin().read_line(&mut input).unwrap();
        let trimmed = input.trim();

        if !trimmed.is_empty() {
            break trimmed.to_string();
        } else {
            println!("\nPlease enter at least one character.\n");
        }
    };

    print!("\nPlease enter a character to count:\n");
    io::stdout().flush().unwrap();

    let mut char_input = String::new();
    io::stdin().read_line(&mut char_input).unwrap();
    let target_char = char_input.trim().chars().next().unwrap_or('\0');

    let count = sentence.chars().filter(|&c| c == target_char).count();

    println!(
        "\nThe character '{}' appears {} time{}.",
        target_char,
        count,
        if count == 1 { "" } else { "s" }
    );
}
